src/core/connection.h

```diff
@@ -8,27 +8,34 @@ namespace mavsdk {
 
 class Connection {
 public:
-    typedef std::function<void(mavlink_message_t& message)> receiver_callback_t;
+    typedef std::function<void(mavlink_message_t& message, Connection* connection)>
+        receiver_callback_t;
 
-    Connection(receiver_callback_t receiver_callback);
+    Connection(receiver_callback_t receiver_callback, bool forward_messages = false);
```

Comment:
I suggest to use an enum for the forwarding option because that will allow to add more rules later and is better in terms of type-safety.

So what about:
```
enum class ForwardingOption {
    ForwardingOff = 0,
    ForwardingOn = 1,
};
```

Reply:
Yep, good point !